main.go

```diff
@@ -2,15 +2,52 @@ package main
 
 import (
 	"fmt"
+	"os"
+	"practice/go-todo-list/app"
+	"practice/go-todo-list/config"
+	"practice/go-todo-list/migration"
 	"practice/go-todo-list/todo"
+
+	"github.com/urfave/cli"
 )
 
 func main() {
-	// config.Load()
+	config.Load()
+	app.InitApp()
+
+	app := cli.NewApp()
+	app.Name = "todo-app"
+	app.Version = "0.0.1"
+	fmt.Println("####DEBUG####")
+
+	app.Commands = []cli.Command{
+		{
+			Name:        "migrate",
+			Description: "Run Database migration",
+			Action: func(c *cli.Context) error {
+				migration.Init()
+				return migration.Up()
+			},
+		},
+		{
+			Name:        "rollback",
+			Description: "Rollback latest database migration",
+			Action: func(c *cli.Context) error {
+				migration.Init()
+				return migration.Down()
+			},
+		},
+	}
 
-	// app.Init()
-	// defer app.Close()
+	if err := app.Run(os.Args); err != nil {
+		fmt.Println("###PANIC YAHA##")
+		panic(err)
+	}
+
+	startApp()
+}
 
+func startApp() {
```

Comment:
we can pull this out in a new file

Reply:
Also, are we not handling the error in lines 65 and 59?

Reply:
delete line 69 as well

Reply:
Also, We are scanning `task_id` from user in `READ` . How would a user know its task_id? It is something generated by us for storing in DB.

Reply:
In the cli app, the user can run show_all to see all the tasks along with their id. This way the user will know the id and will be able to delete the corresponding task. (We obv won't make task string as primary key?)